celesta-sql/src/main/java/ru/curs/celesta/score/discovery/ScoreByScoreResourceDiscovery.java

```diff
@@ -0,0 +1,88 @@
+package ru.curs.celesta.score.discovery;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Enumeration;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import ru.curs.celesta.CelestaException;
+import ru.curs.celesta.score.io.Resource;
+import ru.curs.celesta.score.io.UrlResource;
+
+/**
+ * Implementation of score discovery based on JAR-files resources look up.
+ */
+public final class ScoreByScoreResourceDiscovery implements ScoreDiscovery {
+
+    private static final String SCORE_FILES_LOCATION = "score/score.files";
+
+    @Override
+    public Set<Resource> discoverScore() {
+
+        Map<String, Resource> grainNameToResourceMap = new LinkedHashMap<>();
+
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        try {
+            Enumeration<URL> urls = (classLoader != null)
```

Comment:
Well, I think I have an idea how to test it. 

The only thing that we have to mock here is `Enumeration<URL> urls`. You may either initialize this variable in constructor and then change its value in unit test, or make an overridable getter method and override it in test.

Just invent any set of urls, and then we may check for corner cases like missing score index files and duplicate resources.